### PR TITLE
Add lambda to ship Cloudwatch logs to Opensearch

### PR DESCRIPTION
**Pull request by stafot**
Issue: MM-39482

#### Summary
Add lambda to ship Cloudwatch logs to Opensearch.
The first logs we need to expose are them from Aurora Postgres Clusters

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-39842

### CODE REVIEW

**Review by koox00 — APPROVED** [review]
JavaScript part looks good to me 💪 
Added some nitpicking comments about `let` vs `const` in some parts, but they absolutely don't affect functionality at all

File: lambda-functions/logs-to-opensearch/index.js [hunk 1]
```diff
@@ -0,0 +1,269 @@
+const https = require('https');
```
Comment (spirosoik) [inline lambda-functions/logs-to-opensearch/index.js:1]: I cannot say much here to be honest

File: terraform/aws/modules/logs-to-opensearch/cloudwatch.tf [hunk 2]
```diff
@@ -0,0 +1,31 @@
+
+resource "aws_cloudwatch_metric_alarm" "logs_to_opensearch_errors" {
+  alarm_name          = "Lambda logs-to-opensearch - Errors"
+  alarm_description   = "This lambda ships logs to AWS Opensearch Service. This alarm indicates the lambda had Errors when it was executed."
+  evaluation_periods  = "1"
```
Comment (spirosoik) [inline terraform/aws/modules/logs-to-opensearch/cloudwatch.tf:5]: wondering if all these should configurable and just have default values

File: terraform/aws/modules/logs-to-opensearch/iam.tf [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogGroup",
+                "logs:CreateLogStream",
+                "logs:PutLogEvents"
+            ],
+            "Resource": "arn:aws:logs:*:*:*"
```
Comment (spirosoik) [inline terraform/aws/modules/logs-to-opensearch/iam.tf:38]: should we make it a bit more inclusive?

File: terraform/aws/modules/logs-to-opensearch/variables.tf [hunk 4]
```diff
@@ -0,0 +1,12 @@
+variable "private_subnet_ids" {}
```
Comment (spirosoik) [inline terraform/aws/modules/logs-to-opensearch/variables.tf:1]: please add description to variables

**Review by spirosoik — CHANGES_REQUESTED** [review]
few comments

File: terraform/aws/modules/logs-to-opensearch/iam.tf (hunk 3 above)
Comment (stafot) [reply terraform/aws/modules/logs-to-opensearch/iam.tf:38]: You mean less?

**Review by spirosoik — APPROVED** [review]

**Pull request merged**